src/transactions/transaction.js

```diff
@@ -43,9 +44,12 @@ function createTransaction(recipientId, amount, secret, secondSecret, data, time
 		senderPublicKey: keys.publicKey,
 		timestamp: slots.getTimeWithOffset(timeOffset),
 		asset: {},
-		data,
 	};
 
+	if (data && data.length > 0) {
```

Comment:
I first put another check, but then realized that any error message should be coming from core.